tests/api/test_endpoints.py

```diff
@@ -111,6 +111,37 @@ def test_put_user_update_email__failure(server):
     assert response.status == 400
 
 
+def test_paginate_users__success(server):
+    # add 3 dummy data entries to search for in database.
+    # In total there's 4 with one coming from previous tests.
+    user_info = [[
+        'matt gin', 'ginsstaahh', 'matthewgin10@gmail.com', 'Val1dPassword!'
+    ], ['gin', 'ginsstaahh221', 'matt.gin@hotmail.com', 'Val1dPassword!'],
+                 ['bruno', 'bfcbachman', 'bruno@gmail.com', 'Val1dPassword!']]
+    for full_name, username, email, password in user_info:
+        server.app.test_client.post(
+            '/users',
+            data=json.dumps({
+                'full_name': full_name,
+                'username': username,
+                'email': email,
+                'password': password,
+            }))
+    _, response = server.app.test_client.get('/users/search?size=2')
+    assert response.status == 200
+    body = response.json
+    assert body.get('result_count') == 4
+    assert body.get('page') == 0
+    assert body.get('total_pages') == 2
+
+
+def test_search_users__success(server):
+    _, response = server.app.test_client.get('/users/search?query=gin')
+    assert response.status == 200
+    body = response.json
+    assert len(body.get('results')) == 2
+
```

Comment:
It looks good ! It might be good to add some tests for failure scenarios on your new endpoint for the following cases:
 - queries with large size value
- query does not match any data in the database

Reply:
Right.  A large size value will not break the test though, it will only set pages to have more entries.  And queries that don't have any matches will still return an HTTP status code of 200 however, so it doesn't break.

Reply:
I'll talk to bruno about it though tomorrow

Reply:
Hmm.. I am referring to the APIErrors on line 227 and 233 of bounce/server/api/users.py. It seems like:

- if (size > MAZ_SIZE), client gets a return status 400 with message "size too high"
- if user.search(...) returns a null object, then client gets a return status 404 with message "No users to match query".

Please correct me if I am wrong. We can chat about it tomorrow.